src/commands/force/source/delete.ts

```diff
@@ -133,6 +142,23 @@ export class Delete extends DeployCommand {
     });
     this.componentSet = cs;
 
+    if (sourcepaths) {
+      // determine if user is trying to delete a single file from a bundle, which is actually just an fs delete operation
+      // and then a constructive deploy on the "new" bundle
+      this.components
+        .filter((comp) => {
+          if (comp.type.strategies?.adapter === 'bundle' && comp instanceof SourceComponent) return true;
+        })
+        .map((bundle: SourceComponent) => {
+          sourcepaths.map((sourcepath) => {
+            // walkContent returns absolute paths while sourcepath will usually be relative
```

Comment:
test case for NUTs: delete an entire bundle (sourcepath points to `lwc/someCmp`) does a delete.